// react-thumbnail-player/dist/index.es.js
import React from 'react';
import PropTypes from 'prop-types';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css = ".styles_videoThumbnail__1A0YX {\n  cursor: pointer;\n}\n\n.styles_video__fdeKk {\n  position: relative;\n}\n\n.styles_video__fdeKk .styles_preview__2cXKK {\n  border-radius: 16px;\n}\n\n.styles_video__fdeKk .styles_badge__3dxZ6 {\n  position: absolute;\n  bottom: 16px;\n  right: 16px;\n  background: #613f75;\n  color: #fff;\n  border-radius: 2px;\n  padding: 0 8px;\n  height: 16px;\n  line-height: 16px;\n  font-size: 10px;\n  transition: 0.4s ease-in-out;\n}\n\n.styles_title__1cca0 {\n  padding: 0;\n  margin: 16px 0 8px;\n  font-size: 16px;\n  font-weight: 600;\n  color: #242431;\n}\n\n.styles_views__2l1fl {\n  font-size: 12px;\n  color: #7e7e87;\n}\n";
var styles = { "videoThumbnail": "styles_videoThumbnail__1A0YX", "video": "styles_video__fdeKk", "preview": "styles_preview__2cXKK", "badge": "styles_badge__3dxZ6", "title": "styles_title__1cca0", "views": "styles_views__2l1fl" };
styleInject(css);

var VideoThumbnail = function VideoThumbnail(_ref) {
  var preview = _ref.preview,
      title = _ref.title,
      message = _ref.message,
      badge = _ref.badge,
      badgeBg = _ref.badgeBg,
      muted = _ref.muted,
      width = _ref.width,
      classname = _ref.classname;

  var onHover = function onHover(e) {
    e.target.play();
    e.target.nextElementSibling.style.opacity = 0.4;
  };

  var offHover = function offHover(e) {
    e.target.pause();
    e.target.nextElementSibling.style.opacity = 1;
  };

  return React.createElement(
    'div',
    {
      style: { width: width },
      className: styles.videoThumbnail + ' ' + classname
    },
    React.createElement(
      'div',
      { style: { width: width }, className: styles.video },
      React.createElement('video', {
        onMouseEnter: function onMouseEnter(e) {
          return onHover(e);
        },
        onMouseLeave: function onMouseLeave(e) {
          return offHover(e);
        },
        className: styles.preview,
        src: preview,
        loop: true
        // autoPlay
        , width: width,
        muted: muted
      }),
      React.createElement(
        'div',
        { className: styles.badge, style: { background: badgeBg } },
        badge
      )
    ),
    React.createElement(
      'h4',
      { className: styles.title },
      title
    ),
    React.createElement(
      'span',
      { className: styles.views },
      message
    )
  );
};

VideoThumbnail.propTypes = {
  preview: PropTypes.string,
  title: PropTypes.string,
  message: PropTypes.string,
  badge: PropTypes.string,
  badgeBg: PropTypes.string,
  muted: PropTypes.bool,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  classname: PropTypes.string
};

VideoThumbnail.defaultProps = {
  preview: 'http://clips.vorwaerts-gmbh.de/VfE_html5.mp4',
  title: 'Hi, this is a test title',
  message: '2M Watching Now',
  badge: 'Live',
  // badgeBg: 'lightblue',
  muted: false,
  width: 350,
  classname: ''
};

export default VideoThumbnail;
